src/mpl/configure.ac

```diff
@@ -785,6 +785,78 @@ fi
 ## END OF PTHREAD MUTEX COMMON CHECK
 #######################################################################
 
+#######################################################################
+## START OF ATOMIC CODE
+#######################################################################
+
+PAC_ARG_ATOMIC_PRIMITIVES
+
+AC_DEFUN([MPL_ATOMIC_TEST_PROGRAM], [AC_LANG_PROGRAM([[
+        #include "$1"
+    ]],[[
+        MPL_atomic_int_t a, b;
+        int c;
+
+        MPL_atomic_relaxed_store_int(&a, 0);
+        MPL_atomic_relaxed_store_int(&b, 1);
+        c = MPL_atomic_relaxed_load_int(&a);
+
+        MPL_atomic_release_store_int(&a, 0);
+        MPL_atomic_release_store_int(&b, 1);
+        c = MPL_atomic_acquire_load_int(&a);
+
+        MPL_atomic_fetch_add_int(&a, 10);
+        MPL_atomic_fetch_sub_int(&a, 10);
+
+        c = MPL_atomic_cas_int(&a, 10, 11);
+        c = MPL_atomic_swap_int(&a, 10);
+
+        MPL_atomic_write_barrier();
+        MPL_atomic_read_barrier();
+        MPL_atomic_read_write_barrier();
+    ]])]
+)
+
+dnl MPL_TRY_ATOMIC_HEADER([header file from src/ dir],[symbolic name in configure],[HAVE_ macro suffix],[feature description])
+dnl Does an AC_LINK_IFELSE() to see if the header file works.
+AC_DEFUN([MPL_TRY_ATOMIC_HEADER],[
+if test "$with_mpl_atomic_primitives" = "auto" || \
+   test "$with_mpl_atomic_primitives" = "$2"
+then
+    checked_specified_primitive=yes
+    AC_MSG_CHECKING([for support for $4])
+    SAVE_CFLAGS="$CFLAGS"
+    CFLAGS="$CFLAGS -I${srcdir}/include"
+    AC_LINK_IFELSE([MPL_ATOMIC_TEST_PROGRAM([$1])],
+        [AC_DEFINE([HAVE_$3], [1], [Define to 1 if we have support for $4])]
+        [AC_MSG_RESULT([yes])]
+    ,
+        [AC_MSG_RESULT([no])]
+    )
+    CFLAGS="$SAVE_CFLAGS"
+fi
+])
+
```

Comment:
I think these macro definitions can also move to `confdb/aclocal_cc.m4`, keeping `configure.ac` more concise to read.

Reply:
They are pretty tricky since this macro actually refers to an `mpl_atomic_xxx` header; it is too cumbersome to write comprehensive tests for all the five or six sets of primitives.
I know it's not very clean; if we should detect the availability without using `mpl_atomic_xxx` headers, I will make tests (like https://github.com/pmodels/argobots/blob/master/configure.ac#L612, for example).

Reply:
There are `PAC_PUSH_FLAG` and `PAC_POP_FLAG` to do temporary variable setting. 

I see your point (about why you prefer them in `mpl/configure.ac`), it makes sense.

Reply:
Now the latest one uses `PAC_PUSH_FLAG` and `PAC_POP_FLAG`, but still declare this function in `src/mpl/configure.ac` because of the reason above.